### ISSUE TITLE
Modify positioning of individual parsers in help output

### ISSUE THREAD

**Issue by Jimbo4350** [pcapriotti/optparse-applicative#423]
Is there a way to modify the spacing between the parsers help output? I.E:
![image](https://user-images.githubusercontent.com/9418537/123592176-c9051180-d7e4-11eb-9def-d7264c14634d.png)

This seems to be where the help is rendered: https://github.com/pcapriotti/optparse-applicative/blob/29898f1465139c82a446505627f20e630f671caa/src/Options/Applicative/Help/Core.hs#L239

I'm happy to do the additional work to make this possible.

**Comment by HuwCampbell** [comment]
I'm not exactly sure what you mean. Are you asking if we could add extra mods to particular options, or just generally unhappy with the output?

I think there are probably a few tweaks we could make - e.g., leading pipe separators, an option to show options on the next line with a 2 space hang - but in general it's a hard problem to make something universally pleasing.

**Comment by Jimbo4350** [comment]
@HuwCampbell  Currently the parser help output format immediately after `build-raw` is not modifiable. It would be nice for example to put all of it on a newline rather than have it squashed to the right.

**Comment by dcoutts** [comment]
Right, the point is that the current layout when the command is long, pushes the flags into a very narrow column. That is it looks like:
```
################ ***
                 ***
                 ***
                 ***
                 ***
                 ***
                 ***
```
But it would clearly be much nicer to allow the flags to wrap to the next line when the  and give a layout more like
```
################ \
  **********
  **********
```

**Comment by HuwCampbell** [comment]
Yeah, that's what I was getting at as well.

**Comment by Jimbo4350** [comment]
@HuwCampbell I'm happy to do the work (with some pointers from you) if you are too busy too :smile:

**Comment by HuwCampbell** [comment]
Sure.

You would have to look [here](https://github.com/pcapriotti/optparse-applicative/blob/29898f1465139c82a446505627f20e630f671caa/src/Options/Applicative/Help/Core.hs#L266-L272).

The idea is to add a parser preference for hanging instead of aligning.

**Comment by HuwCampbell** [comment]
I've just released 0.17 which should fix up some of these issues.
